demos/common/models/src/hpe_model_openpose.cpp

```diff
@@ -0,0 +1,220 @@
+}
+
+std::shared_ptr<InternalModelData> HPEOpenPose::preprocess(const InputData& inputData, InferenceEngine::InferRequest::Ptr& request) {
+    auto& image = inputData.asRef<ImageInputData>().inputImage;
+    Blob::Ptr frameBlob = request->GetBlob(inputsNames[0]);
+    cv::Mat resizedImage;
+    double scale = inputLayerSize.height / static_cast<double>(image.rows);
+    cv::resize(image, resizedImage, cv::Size(), scale, scale, cv::INTER_CUBIC);
+    int h = resizedImage.rows;
+    int w = resizedImage.cols;
+    if (inputLayerSize.width < w)
+        throw std::runtime_error("The image aspect ratio doesn't fit current model shape");
+    cv::Mat paddedImage;
+    pad(1) = inputLayerSize.height - h;
+    pad(3) = inputLayerSize.width - w;
+    cv::copyMakeBorder(resizedImage, paddedImage, pad(0), pad(1), pad(2), pad(3),
+                       cv::BORDER_CONSTANT, meanPixel);
+    matU8ToBlob<uint8_t>(paddedImage, frameBlob);
+    return std::shared_ptr<InternalModelData>(new InternalImageModelData(image.cols, image.rows));
+}
+
+std::unique_ptr<ResultBase> HPEOpenPose::postprocess(InferenceResult & infResult) {
+    HumanPoseResult* result = new HumanPoseResult;
+    *static_cast<ResultBase*>(result) = static_cast<ResultBase&>(infResult);
+
+    auto outputMapped = infResult.outputsData[outputsNames[0]];
+    auto heatMapsMapped = infResult.outputsData[outputsNames[1]];
+
+    const SizeVector& outputDims = outputMapped->getTensorDesc().getDims();
+    const SizeVector& heatMapDims = heatMapsMapped->getTensorDesc().getDims();
+
+    float* predictions = outputMapped->rmap().as<float*>();
+    float* heats = heatMapsMapped->rmap().as<float*>();
+
+    const auto& internalData = infResult.internalModelData->asRef<InternalImageModelData>();
+
+    std::vector<cv::Mat> heatMaps(keypointsNumber);
+    for (size_t i = 0; i < heatMaps.size(); i++) {
+        heatMaps[i] = cv::Mat(heatMapDims[2], heatMapDims[3], CV_32FC1,
+                              reinterpret_cast<void*>(
+                                  const_cast<float*>(
+                                      heats + i * heatMapDims[2] * heatMapDims[3])));
+    }
+    resizeFeatureMaps(heatMaps);
+
+    std::vector<cv::Mat> pafs(outputDims[1]);
+    for (size_t i = 0; i < pafs.size(); i++) {
+        pafs[i] = cv::Mat(heatMapDims[2], heatMapDims[3], CV_32FC1,
+                          reinterpret_cast<void*>(
+                              const_cast<float*>(
+                                  predictions + i * heatMapDims[2] * heatMapDims[3])));
+    }
+    resizeFeatureMaps(pafs);
+
+    std::vector<HumanPose> poses = extractPoses(heatMaps, pafs);
+
+    cv::Size fullFeatureMapSize = heatMaps[0].size() * stride / upsampleRatio;
+    float scaleX = internalData.inputImgWidth /
+            static_cast<float>(fullFeatureMapSize.width - pad(1) - pad(3));
+    float scaleY = internalData.inputImgHeight /
+            static_cast<float>(fullFeatureMapSize.height - pad(0) - pad(2));
```

Comment:
It looks it's incorrect to have padding affecting scale values. `scaleX` and `scaleY` correspond to scale introduced by the net during inference. That scaling doesn't care if the pixels represent an actual image or they are just padded. That removes the need in `pad` in postprocessing, so you shouldn't store it in `HPEOpenPose`.

Reply:
Done.